src/arithmetic/boolean_funcs/boolean_funcs.c

```diff
@@ -147,6 +139,8 @@ SIValue AR_EQ(SIValue *argv, int argc) {
 	int disjointOrNull = 0;
 	int res = SIValue_Compare(a, b, &disjointOrNull);
 	if(disjointOrNull == COMPARED_NULL) return SI_NullVal();
+	if(disjointOrNull == COMPARED_NAN) return SI_BoolVal(false);
```

Comment:
@AviAvni Just making sure, if both element are NAN we treat them as not equal, is this intended?
This question is also relevant to the rest of the functions.

Reply:
yes this is checked in the TCK tests